tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -1474,6 +1475,44 @@ public:
    size_t GetAlignment() const final { return alignof(std::unique_ptr<char>); }
 };
 
+class ROptionalField : public RNullableField {
+   class ROptionalDeleter : public RDeleter {
+   private:
+      std::unique_ptr<RDeleter> fItemDeleter;
+
+   public:
+      explicit ROptionalDeleter(std::unique_ptr<RDeleter> itemDeleter) : fItemDeleter(std::move(itemDeleter)) {}
+      void operator()(void *objPtr, bool dtorOnly) final;
+   };
+
+   std::unique_ptr<RDeleter> fItemDeleter;
+
+   /// Given a pointer to an std::optional<T> in from, extract a pointer to the value T* and a pointer
+   /// to the engagement boolean. Assumes that an std::optional<T> is stored as
+   /// `struct { T t; bool engagement; };`
```

Comment:
Can there be cases where this is not true? Can/should we check for this?

Reply:
We rely on the unit tests to catch when this is not the case. I added some more testing on the object sizes and alignments.

Reply:
It is likely not true for the (unlikely) `std::optional<std::variant<...>>` on MacOS (i.e. it likely has the same padding issue as the converse).

Reply:
It doesn't, the class inheritance structure is slightly different in this case.